_posts/2022-05-17-calldata-reencode-size-check-bug.md

```diff
@@ -0,0 +1,72 @@
+---
+layout: post
+published: true
+title: 'Size Check Bug in Nested Calldata Array ABI-Reencoding'
+date: '2022-05-17'
+author: Solidity Team
+category: Security Alerts
+---
+
+On April 7, 2022, a bug in the Solidity code generator was reported by
+John Toman of the Certora development team. Certora's bug disclosure post can be found [here](https://medium.com/certora/incorrect-calldata-validation-in-inter-contract-communication-certora-bug-disclosure-8a467ce92ca3).
+
+The bug is fixed with [Solidity version 0.8.14](https://github.com/ethereum/solidity/releases/tag/v0.8.14)
+released on May 17, 2022. The bug was first introduced in Solidity version 0.5.8.
+
+We assigned the bug a severity of "very low".
+
+## Which Contracts are Affected?
+
+You might be affected if you pass a nested array directly to another external
+function call or use ``abi.encode`` on it.
+
+If calldata is malformed in a certain way, the call might not revert as it should.
+Instead, data with zeros is passed on to the called contract or to the ``abi.encode`` function.
+
+## Technical Details
+
+Calldata validation for nested dynamic types is deferred until the first access
+to the nested values. Such an access may for example be a copy to memory or an
+index or member access to the outer type. While in most such accesses calldata
+validation correctly checks that the data area of the nested array is completely
```

Comment:
This assumes that the reader knows what a "data area" of a dynamically encoded array is. Maybe we whould link to the ABI encoding page in the docs?

Reply:
Added a link.